### PR TITLE
add OAuth example to readme (#219)

### PR DESCRIPTION
**Pull request by thisconnect**

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -250,6 +250,39 @@ got.post('google.com', {
...
+});
+
+const token = {
+  public: ACCESS_TOKEN,
+  secret: ACCESS_TOKEN_SECRET
+};
+
+const request_data = {
+	url: 'https://api.twitter.com/1.1/statuses/home_timeline.json',
+	method: 'GET'
```
Comment (sindresorhus) [inline readme.md]: This is moot as it's already the default.

File: readme.md [hunk 2]
```diff
@@ -250,6 +250,39 @@ got.post('google.com', {
...
+  signature_method: 'HMAC-SHA1'
+});
+
+const token = {
+  public: ACCESS_TOKEN,
+  secret: ACCESS_TOKEN_SECRET
+};
+
+const request_data = {
+	url: 'https://api.twitter.com/1.1/statuses/home_timeline.json',
```
Comment (sindresorhus) [inline readme.md]: The URL can just be a string argument to `got`.

File: readme.md [hunk 3]
```diff
@@ -250,6 +250,39 @@ got.post('google.com', {
...
+	method: 'GET'
+};
+
+got(request_data.url, {
```
Comment (sindresorhus) [inline readme.md]: You need to specify the `json: true` option

File: readme.md [hunk 4]
```diff
@@ -250,6 +250,39 @@ got.post('google.com', {
 ```
 
 
+## OAuth
+
+You can use the [`oauth-1.0a`](https://www.npmjs.com/package/oauth-1.0a) module to create a signed OAuth request:
```
Comment (sindresorhus) [inline readme.md]: Link to the GitHub repo

File: readme.md [hunk 5]
```diff
@@ -250,6 +250,39 @@ got.post('google.com', {
...
+
+```js
+const got = require('got');
+const OAuth = require('oauth-1.0a');
+
+
+const oauth = OAuth({
+  consumer: {
+    public: CONSUMER_KEY,
+    secret: CONSUMER_SECRET
```
Comment (sindresorhus) [inline readme.md]: Unclear where these variables (`CONSUMER_SECRET`) are coming from. Use `process.env.CONSUMER_SECRET` for all of these.

File: readme.md (hunk 2 above)
Comment (thisconnect) [inline readme.md]: Just had that because the url is used in 2 places. I'll update...

File: readme.md [hunk 6]
```diff
@@ -250,6 +250,39 @@ got.post('google.com', {
...
+  },
+  signature_method: 'HMAC-SHA1'
+});
+
+const token = {
+  public: ACCESS_TOKEN,
+  secret: ACCESS_TOKEN_SECRET
+};
+
+const request_data = {
```
Comment (sindresorhus) [inline readme.md]: Should be camelCase

File: readme.md (hunk 2 above)
Comment (sindresorhus) [inline readme.md]: Yeah, just noticed. Never mind.

Comment (thisconnect) [inline readme.md]: How about a url variable instead of request_data?

File: readme.md (hunk 4 above)
Comment (thisconnect) [inline readme.md]: I copied that from the previous example. I am going to change form-data to link to github too.

File: readme.md (hunk 2 above)
Comment (sindresorhus) [inline readme.md]: `requestData` would be ok.

**Comment by thisconnect** [comment]
@sindresorhus I'll close this and open a new PR (squashed) ok?

**Comment by julien-f** [comment]
@thisconnect, @sindresorhus can squash on merge, I don't think you have to create a new PR.

**Comment by thisconnect** [comment]
nice to know, thanks :)

**Comment by thisconnect** [comment]
BTW: option.query can not be used with OAuth as the query has to be part of the requestData.url in oauth.authorize

**Comment by thisconnect** [comment]
alternative:

```js
const got = require('got');
const OAuth = require('oauth-1.0a');

const oauth = OAuth({
    consumer: {
        public: process.env.CONSUMER_KEY,
        secret: process.env.CONSUMER_SECRET
    },
    signature_method: 'HMAC-SHA1'
});

const token = {
    public: process.env.ACCESS_TOKEN,
    secret: process.env.ACCESS_TOKEN_SECRET
};

const url = 'https://api.twitter.com/1.1/statuses/home_timeline.json';

got(url, {
    headers: oauth.toHeader(oauth.authorize({ url: url, method: 'GET' }, token)),
    json: true
});
```

**Comment by sindresorhus** [comment]
👍 

`{ url: url, method: 'GET' }` => `{url, method: 'GET'}`

File: readme.md [hunk 7]
```diff
@@ -250,6 +250,36 @@ got.post('google.com', {
...
+
+const token = {
+	public: process.env.ACCESS_TOKEN,
+	secret: process.env.ACCESS_TOKEN_SECRET
+};
+
+const url = 'https://api.twitter.com/1.1/statuses/home_timeline.json';
+
+got(url, {
+	headers: oauth.toHeader(oauth.authorize({ url, method: 'GET' }, token)),
```
Comment (SamVerschueren) [inline readme.md]: Remove spaces inside the object `{url, method: 'GET'}`

**Comment by sindresorhus** [comment]
Looks great! Thanks @thisconnect :)

**Comment by thisconnect** [comment]
Thank you too!!

**Pull request merged**